warehouse/models/intermediate/gtfs_quality/guidelines_checks/int_gtfs_quality__pathways_valid.sql

```diff
@@ -57,30 +30,88 @@ validation_fact_daily_feed_codes_pathway_related AS (
             )
 ),
 
-pathway_validation_notices_by_day AS (
+feed_has_rail AS (
+    SELECT
+        feed_key,
+        LOGICAL_OR(route_type = "2") AS has_rail
+    FROM dim_routes
+    GROUP BY 1
+),
+
+feed_has_stations AS (
     SELECT
         feed_key,
-        SUM(total_notices) as validation_notices
-    FROM validation_fact_daily_feed_codes_pathway_related
+        LOGICAL_OR(parent_station IS NOT NULL) AS has_parent_station,
+        LOGICAL_OR(LOWER(stop_name) LIKE '%station%' OR LOWER(stop_name) LIKE '%transit center%') AS has_station_name
+    FROM dim_stops
     GROUP BY 1
 ),
 
+feed_has_pathways AS (
+    SELECT
+        feed_key,
+        LOGICAL_OR(gtfs_filename = "pathways") AS has_pathways
+    FROM files
+    GROUP BY 1
+),
+
+pathways_eligibile AS (
+    SELECT
+        COALESCE(rail.feed_key, stations.feed_key) AS feed_key,
+        COALESCE(has_parent_station, FALSE)
+            OR COALESCE(has_station_name, FALSE)
+            OR COALESCE(has_rail, FALSE)
+            AS is_pathways_eligible,
+        has_parent_station,
+        has_rail,
+        has_station_name
+    FROM feed_has_rail AS rail
+    FULL OUTER JOIN feed_has_stations AS stations
+        ON rail.feed_key = stations.feed_key
+),
+
+feed_pathways_notices AS (
+    SELECT
+        feed_key,
+        SUM(total_notices) AS validation_notices
+    FROM pathways_notices
+    GROUP BY 1
+),
+
+check_start AS (
+    SELECT MIN(date) AS first_check_date
+    FROM pathways_notices
+),
+
 int_gtfs_quality__pathways_valid AS (
     SELECT
-        t1.date,
-        {{ pathways_valid() }} AS check,
-        {{ accurate_accessibility_data() }} AS feature,
-        t1.feed_key,
+        idx.* EXCEPT(status),
+
+        is_pathways_eligible,
+        has_parent_station,
+        has_rail,
+        has_pathways,
+        has_station_name,
+        validation_notices,
         CASE
-            WHEN t2.feed_key IS null THEN {{ guidelines_na_check_status() }}
-            WHEN t3.validation_notices = 0 THEN {{ guidelines_pass_status() }}
-            WHEN t3.validation_notices > 0 THEN {{ guidelines_fail_status() }}
+            WHEN has_schedule_feed
+                THEN
+                    CASE
+                        WHEN has_pathways AND validation_notices = 0 AND is_pathways_eligible THEN {{ guidelines_pass_status() }}
+                        WHEN idx.date < first_check_date THEN {{ guidelines_na_too_early_status() }}
+                        WHEN NOT COALESCE(is_pathways_eligible, FALSE) OR (feed_pathways_notices.feed_key IS NULL) THEN {{ guidelines_na_check_status() }}
+                        WHEN (validation_notices > 0 OR NOT has_pathways) AND is_pathways_eligible THEN {{ guidelines_fail_status() }}
```

Comment:
super mega nit: could you put `is_pathways_eligible` first in the lines where it's used? since it's the first "filter" it seems

Reply:
done